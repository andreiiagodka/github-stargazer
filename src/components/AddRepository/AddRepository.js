import React from 'react'
import PropTypes from 'prop-types'

import { withRouter } from 'react-router'

import Layout from '../UI/Layout/Layout'
import Header from '../UI/Header/Header'
import Body from './Body/Body'

import { getRepositoryData } from '../../shared/github'
import { createRepository } from '../../shared/firebase'

const AddRepository = ({ history }) => {
  const header = <Header title='Add Repository' />
  const body = <Body 
    getRepositoryData={getRepositoryData} 
    createRepository={createRepository}
    history={history} />
    
  return <Layout header={header} body={body} />
}

Body.propTypes = {
  history: PropTypes.object.isRequired
}

export default withRouter(AddRepository)
