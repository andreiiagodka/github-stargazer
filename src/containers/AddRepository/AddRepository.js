import React, { Component, Fragment } from 'react'
import { Card } from 'react-bootstrap'
import { Redirect} from 'react-router';

import { connect } from 'react-redux';
import * as actions from '../../store/actions/actions';

import Spinner from '../../components/UI/Spinner/Spinner';
import Header from '../../components/UI/Header/Header'
import Form from '../../components/AddRepository/Form/Form'

class createRepository extends Component {
  submitHandler = event => {
    event.preventDefault()

    this.props.createRepository(event.target.elements.name.value)
  }

  render() {
    if (this.props.redirectTo) {
      return <Redirect to={this.props.redirectTo} />
    }

    let content = <Spinner />
    if (!this.props.loading) {
      content = (
        <Fragment>
          <Card.Header>
            <Header title='Add Repository' />
          </Card.Header>
          <Card.Body>
            <Form error={this.props.error} submitHandler={this.submitHandler} />
          </Card.Body>
        </Fragment>
      )
    }

    return content
  }
}

const mapStateToProps = state => {
  return {
    loading: state.createRepository.loading,
    error: state.createRepository.error,
    redirectTo: state.createRepository.redirectTo
  }
}

const mapDispatchToProps = dispatch => {
  return {
    createRepository: (name) => dispatch(actions.createRepository(name))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(createRepository)