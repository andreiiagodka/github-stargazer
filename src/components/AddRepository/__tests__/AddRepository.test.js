import React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router'

import AddRepository from '../AddRepository'

describe('<AddRepository /> component', () => {
  it('renders component correctly', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <AddRepository />
      </MemoryRouter>
    ).toJSON()
  
    expect(tree).toMatchSnapshot()
  })
})
