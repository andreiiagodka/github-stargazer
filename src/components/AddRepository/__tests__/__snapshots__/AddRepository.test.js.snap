// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddRepository /> component renders component correctly 1`] = `
Array [
  <div
    className="card-header"
  >
    <div
      className="row"
    >
      <div
        className="col-md-2"
      />
      <div
        className="text-center col-md-8"
      >
        <h5>
          Add Repository
        </h5>
      </div>
      <div
        className="text-right col-md-2"
      />
    </div>
  </div>,
  <div
    className="card-body"
  >
    <div
      className="container-fluid"
    >
      <div
        className="col-md-8 offset-md-2"
      >
        <form
          className=""
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <input
              className="form-control"
              id="name"
              name="name"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Enter repository name, e.g. rubygarage/truemail"
              value=""
            />
          </div>
          <button
            className="btn btn-primary btn-block btn-lg"
            disabled={false}
            type="submit"
          >
            Add
          </button>
        </form>
      </div>
    </div>
  </div>,
]
`;
